conduct.md

```diff
@@ -11,9 +11,10 @@ redirect_from:
 ## Scala Code of Conduct
 
 We are committed to providing a friendly, safe and welcoming environment for
-all, regardless of level of experience, gender, gender identity and expression,
-sexual orientation, disability, personal appearance, body size, race, ethnicity,
-age, religion, nationality, or other such characteristics.
+all, regardless of age, body size, disability, ethnicity, sex characteristics,
+gender identity and expression, level of experience, education, socio-economic
+status, nationality, personal appearance, race, religion, or sexual identity
+and orientation.
 
```

Comment:
:+1: to aligning with Contributor Covenant here

Reply:
What about aligning it with https://www.fossi-foundation.org/code-of-conduct ?
all, regardless of age, culture, ethnicity, genotype, gender identity or expression, language, national origin, neurotype, phenotype, political beliefs, profession, race, religion, sexual orientation, socioeconomic status, subculture and technical ability.

Reply:
@soronpo can you explain a bit more why you think that would be better?

Reply:
To add a bit of background, the currently published text in this section appears to be derived from [Contributor Covenant v1.3](https://www.contributor-covenant.org/version/1/3/0/code-of-conduct):

> We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, or nationality.

So, my aim was to align with the text in the latest version of the same document.

Reply:
I'm not terribly thrilled about the upstream document.  If it's alphabetized, it's full of mistakes.  If it's in order of importance or how often it comes up, it seems really bizarre.  Alignment is good, I suppose, but this looks like we don't care.  The original list seems better ordered, and I know what all the terms mean.  It looks like they replaced "gender" with "sex characteristics", I guess because technically the term is "sex" but that's ambiguous between the morphology (which is what is meant), and the act?  Also, there are no longer "other such characteristics", so for example we're absolutely able to be unfriendly, dangerous, and unwelcoming to people on the basis of whether or not they like cats?

Anyway, I don't imagine the change will cause problems, but I think it's strictly worse.

Reply:
@soronpo on the one hand, I'm a bit reluctant to see the Scala CoC become _even more_ of a Frankenstein, with bits and pieces assembled from _even more_ other codes. on the other hand, the FOSSi wording does seem higher quality (and Rex has done a good job of unpacking some of the reasons)

I'm fine with any of these versions (including just sticking with what we originally had)

Reply:
It seems both lists align in most ways. The notable exception is `political beliefs` that has no equivalent in the Contributor Covenant. I believe that is on purpose, and not in a good way.

Reply:
ah, I see. in that case I think using the FOSSi text would be a more substantial change that would need a separate PR and discussion. I suggest we keep this PR scoped more modestly.

Reply:
Both the Fossi code and the Contributor Covenant explicitly welcome _everyone_, with no exceptions.  The lists, in both cases, are just to warn people that one recognizes these things come up and that, as with every other reason, these things will not be stood for.  So I don't have much of a preference for content, as long as it covers the basics.  If we're arguing about whether a particular unpleasantness was or wasn't called out by name, things have already gone awry.

If the "for all" bit is rhetorical flourish, not intended to be taken seriously, while the list is intended to be taken seriously, then I wouldn't think "political beliefs" is the biggest difference.  People don't talk about politics much on any of these venues; it's off-topic.

"language" seems more likely to come up (e.g. is a venue English-only, and if so, what is your attitude towards people who don't respect that...or people who are upset when non-English is used extensively in a venue without a specified language).  "neurotype" is also potentially significant (again, both ways).

Again, I don't think any of these things ought to matter much if we forthrightly mean "all", "everyone", and the like.  If we mean "we feel free to make things unpleasant for Russian-speaking autistic Libertarians", we probably ought to change the "all" stuff before worrying about Fossi vs. Contributor Covenant.

Reply:
@Ichoran I agree, "for all" is the most important part of the statement.

I can see now that "other such characteristics" wasn't in the original Contributor Covenant document, and seems to be adapted from the slightly different version in the [Rust Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct):

> We are committed to providing a friendly, safe and welcoming environment for all, regardless of level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar characteristic.

Removing "or other such characteristics" was an unintended oversight on my part. I'll put it back.

As for the rest of it, I don't have a strong preference for the new vs. old wording, but this did come up from internal feedback when drafting the Lightbend variant at https://www.lightbend.com/conduct. Some felt that the list was missing characteristics that we should explicitly include. A list like this can never be fully comprehensive, but I am very reluctant to remove characteristics from the list that has already been published there, as it might be incorrectly interpreted to mean we no longer consider those characteristics to be protected.

So if the consensus here is not to change this section, I'll remove that from this pull request, but leave it as a difference between the Scala and Lightbend versions of the document.

On the other hand, if the decision is to reorder or add additional characteristics, I'd want to align the Lightbend copy to it.

Reply:
> So if the consensus here is not to change this section, I'll remove that from this pull request, but leave it as a difference between the Scala and Lightbend versions of the document.

Okay. That seems... inelegant, but not terrible. I guess if anyone asks, we can point them at this discussion and shrug.

Reply:
> we can point them at this discussion

and/or at https://www.contributor-covenant.org/faq ("The Contributor Covenant explicitly lists a set of protected classes; does this make it acceptable to discriminate or make others feel unwelcome based on other factors?" ...)

Reply:
> It looks like they replaced "gender" with "sex characteristics", I guess because technically the term is "sex" but that's ambiguous between the morphology

hi, your local trans girl here 👋

gender (identity) and sex are not the same. the former is very important to a lot of people (myself included), and in very simplified terms is "are you nonbinary, a woman, genderqueer, a man, genderfluid, bigender, etc.?". the latter is... complicated for a variety of reasons. I don't want to discount the identity or experience of intersex folks, but for the most part, sex (in western culture) is garbage and an attempt to classify everyone as "male" or "female", regardless of the fact that those categories don't fit well and aren't useful for a lot of people.

tl;dr: "gender" and "sex" are not the same. IMO, gender is the more important one (please please *please* do not change the wording to "sex"), but "sex characteristics" may also be a useful category to explicitly specify (I'm not intersex, so I can't really speak for those who are).

Reply:
Thanks for adding your perspective, @NthPortal!

I'd also like to point out that the new list includes: "gender identity and expression". So, I don't believe "gender" was simply replaced by "sex characteristics", but rather the new text acknowledges inclusivity across both of the distinct categories of gender and sex.

Reply:
Here are the origins of the upstream changes. The addition of "sex characteristics" and the removal of "gender" were made in separate changes:

- Collapsing "gender, gender expression and identity" to only "gender expression and identity" was done in the initial draft of the 1.4 version: https://github.com/ContributorCovenant/contributor_covenant/pull/226. There's no explicit explanation of the change.
- Adding "sex characteristics" came in a later update with this explanation: "Adding 'sex characteristics' to the examples, as some harassment and discrimination is based on sex characteristics, including but not limited to references to menstruation, fertility, and genitalia." https://github.com/ContributorCovenant/contributor_covenant/pull/550

Reply:
@TimMoore sorry, I wasn't objecting to the proposed wording; I was replying to Rex's comment.

I'm personally happy with just straight aligning with the Contributor Covenant here. They have a good list, and it makes maintenance more straightforward.

Reply:
I think the changes are fine!

Also, if I object to a certain line of reasoning, it doesn't follow that I object to the changes.

As a biologist (hey, I'm your local biologist here! :wave:) who understands the regulation of sex-specific expression in mammals (and the rates of morphology that isn't clearly dimorphic in one direction or the other), I'm rather baffled by the apparent claim that classifying _almost_ everyone (~99.8%) as "male" or "female" morphologically (sans hormone therapy) isn't the only rational thing to do.  I'm also baffled by the apparent culture-bashing.  Telling people that their culture has aspects which are "garbage" isn't a very good way to be inclusive.  When something has problems, pointing out the flaws politely and constructively is generally the way to go?

But none of that matters to the CoC.  The point of the CoC is to call us to be kind and welcoming, and to specifically reject that particular kind of nastiness that often manifests when someone seems to be "different" in some substantive way.  Thanks, Tim, for digging out the justification!  I had assumed they actually just meant "sex" since it seems weird to me to not mention sex but mention sex characteristics, but sure! (Background: with the fuss over bathrooms and other semi-private places, the objection is generally voiced as something like _X is a man and I don't care how they self-identify._ It's specifically sex, not gender, not sex characteristics (isasmuch as those can be divorced from sex).)

Anyway--I'm also fine with just aligning with CoC!

Reply:
I feel like this has become contentious in a way that's outside the boundaries of this PR, so I'd like to ask that any further background discussion about the deeper underlying issues here go elsewhere.

It sounds like everyone is at least reasonably comfortable with adopting the Contributor Covenant language, or at least doesn't consider it notably worse than what we have already, so that seems like consensus to me — within the boundaries of this PR anyway, which is where I ask that we stay.